use strict;
use warnings;


# This script is used to count how many transcripts were mapped to the reference genome based on a coverage cutoff
# Usage: Perl 

##############################################################################################

my %tran_lens; #To store the lengths of raw transcripts 

open In, "$ARGV[0]" or die "$!"; # This file obtains from iTools Fatools stat for fasta file

while (<In>) {
next if ($_=~/\#/);
my @temp = split (/\t/,$_);
$tran_lens{$temp[0]} = $temp[1];
}
print "$tran_lens{Acropora_millepora_20477}\n";

close In;

##############################################################################################

$/ = "expressed_sequence_match";

open Gff, "$ARGV[1]" or die "$!"; # This file is the est2genome gff output from maker generated by exonerate

while (<Gff>) {
if ($_=~/Acropora_millepora/) {
my $len = 0;
next if (! ($_ =~/match_part/));
my @exon_match = split (/\n/,$_);
my $first = shift @exon_match;
$first =~ /(.*)ID\=(.*)\;Name\=(.*)/;
my $name = $3;

pop(@exon_match);
foreach my $exon (@exon_match) {
$exon =~/(.*)\;Target=Acropora_millepora_(\d+) (\d+) (\d+) (.*)\;Gap\=M(\d+)/;
my $exon_len = $4-$3+1;
$len = $len + $exon_len;
}
my $rate = $len/$tran_lens{$name};
print "$name\t$len\t$rate\n";
}
}
